src/debug/main.c

```diff
@@ -28,8 +30,29 @@ int	check_enter_space(char *input)
 	return (1);
 }
 
+/**
+ * @brief	Runs the minishell.
+ * 
+ * Signal lines:
+ * 1. SIGINT: Sets up the signal handler for SIGINT (Ctrl + C), so that 
+ * minishell displays a new prompt when the signal is received.
+ * 2. SIGQUIT: Ignores SIGQUIT (Ctrl + \), so that minishell doesn't quit when
+ * the signal is received.
+ * 3. global_signal: Is set to 0 to indicate that the heredoc process has not
+ * been run (and may not even be) yet for the current command, and so
+ * the execution will be run unless the heredoc process is interrupted with
+ * SIGINT (Ctrl + C) when (and if) it runs.
+ * 
+ * The global variable g_signal is checked before the execution is launched
+ * because in case it isn't 0 (but SIGUSR1), it indicates that the heredoc
+ * process was interrupted with SIGINT (Ctrl + C), and so the whole command
+ * should be canceled and a new prompt displayed.
+*/
 int	minishell_loop(t_data *data)
 {
+	signal(SIGINT, handle_sigint);
+	signal(SIGQUIT, SIG_IGN);
+	g_signal = 0;
```

Comment:
Global variable set again as it may get changed during the run of the program and each new prompt needs to have it reset to 0.